components/brave_wallet_ui/components/desktop/popup-modals/add-account-modal/hardware-wallet-connect/accounts-list.tsx

```diff
@@ -207,10 +207,13 @@ function AccountListItem (props: AccountListItemProps) {
 
   React.useMemo(() => {
     getBalance(account.address, account.coin).then((result) => {
-      const formattedBalance = new Amount(result)
-        .divideByDecimals(selectedNetwork.decimals)
-        .format()
-      setBalance(formattedBalance)
+      let amount = new Amount(result)
+      if (account.coin === BraveWallet.CoinType.SOL) {
+        amount = amount.divideByDecimals(9)
+      } else {
+        amount = amount.divideByDecimals(selectedNetwork.decimals)
+      }
+      setBalance(amount.format())
```

Comment:
I made this modification because I saw the balances were incorrect in the 'Import hardware account' modal.  The reason they were incorrect was because I still had mainnet Ethereum as the selected network.  This resolved this issue, but I don't know if this is the correct approach.

Reply:
Seems okay to me!